store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReader.java

```diff
@@ -74,6 +74,8 @@ public boolean hasNext() throws IOException, InterruptedException {
         return false;
       } else {
         index++;
+        // current reader is closed
+        currentReader.close();
```

Comment:
Shall we close this in a finally block? else still there will be a chance of memory leak.